File: Assets/Convergence Assets/Scripts/XRidge.js
class XRidge extends System.Object
{
	var myMeshObject : GameObject;
	var myLabel : GameObject;
	var trans : Transform;
	var myMesh : Mesh;
	var vertexColors : float[];
	var myRenderer : Renderer;
	var myBin : int;
	var myRow : int;
	
	function AddRidge(aMeshObject : GameObject, passColors : float[], amesh : Mesh, bin : int, row : int)
	{
		myMeshObject = aMeshObject;
		trans = aMeshObject.transform;
		myRenderer = aMeshObject.GetComponent.<Renderer>();
		myMesh = amesh;
		vertexColors = passColors;
		myBin = bin;
		myRow = row;
	}
	
	function AddRidge(aMeshObject : GameObject, amesh : Mesh, bin : int, row : int)
	{
		myMeshObject = aMeshObject;
		trans = aMeshObject.transform;
		myRenderer = aMeshObject.GetComponent.<Renderer>();
		myMesh = amesh;
		myBin = bin;
		myRow = row;
	}
	
	function AddLabel(alabel : GameObject)
	{
		myLabel = alabel;
	}
	
	function PositionRidge(location : Vector3, extent : Vector3, numSlots : int, whichSlot : int)
	{
		var newWidth = extent.z / numSlots;
		var posOff = newWidth * whichSlot;
		trans.position = location + Vector3(0, 0, posOff);
		trans.localScale = extent;
		trans.localScale.z /= numSlots;
	}
	
	function NewHeight(nh : float)
	{
		trans.localScale.y = nh;
	}
	
	function Show(doShow : boolean)
	{
		myRenderer.enabled = doShow;
	}
	
	function AdvanceUV(newUV : Vector2, adv : Vector2)
	{
		var uv : Vector2[] = myMesh.uv;
		var i : int;
		for(i = 0; i < uv.length; ++i)
		{
			var myuv : Vector2 = newUV + (adv * i);
			uv[i] += myuv;
		}
		myMesh.uv = uv;
	}
	
	function ReleaseRidge()
	{
		Debug.Log("Destroying in ReleaseRidge");
//				gameObject.Destroy(myMeshObject.gameObject);
	}
}

File: Assets/Convergence Assets/Scripts/XRidge.cs
using UnityEngine;
using System.Collections;

[System.Serializable]
///<summary>
///Stauffer: a 'ridge' is a mesh and assoc'ed objects that visualizes a full row of data
/// </summary>
public class XRidge : object
{
    public GameObject myMeshObject;
    public GameObject myLabel;
    public Transform trans;
    public Mesh myMesh;
    public float[] vertexColors;
    public Renderer myRenderer;
    public int myBin;
    public int myRow;
    /* Stauffer - unused 
    public virtual void AddRidge(GameObject aMeshObject, float[] passColors, Mesh amesh, int bin, int row)
    {
        this.myMeshObject = aMeshObject;
        this.trans = aMeshObject.transform;
        this.myRenderer = aMeshObject.GetComponent<Renderer>();
        this.myMesh = amesh;
        this.vertexColors = passColors;
        this.myBin = bin;
        this.myRow = row;
    }*/

    public virtual void AddRidge(GameObject aMeshObject, Mesh amesh, int bin, int row)
    {
        this.myMeshObject = aMeshObject;
        this.trans = aMeshObject.transform;
        this.myRenderer = aMeshObject.GetComponent<Renderer>();
        this.myMesh = amesh;
        this.myBin = bin;
        this.myRow = row;
    }

    public virtual void AddLabel(GameObject alabel)
    {
        this.myLabel = alabel;
    }

    /* Stauffer - unused 
    public virtual void PositionRidge(Vector3 location, Vector3 extent, int numSlots, int whichSlot)
    {
        float newWidth = extent.z / numSlots;
        float posOff = newWidth * whichSlot;
        this.trans.position = location + new Vector3(0, 0, posOff);
        this.trans.localScale = extent;

        {
            float _57 = this.trans.localScale.z / numSlots;
            Vector3 _58 = this.trans.localScale;
            _58.z = _57;
            this.trans.localScale = _58;
        }
    }*/

    public virtual void NewHeight(float nh)
    {

        {
            float _59 = nh;
            Vector3 _60 = this.trans.localScale;
            _60.y = _59;
            this.trans.localScale = _60;
        }
    }

    //Stauffer - used (at least in one way) to show/hide ridges based on their bin number, and what bins are chosen for viewing
    public virtual void Show(bool doShow)
    {
        this.myRenderer.enabled = doShow;
    }

    //Stauffer - seems unused
    public virtual void AdvanceUV(Vector2 newUV, Vector2 adv)
    {
        int i = 0;
        Vector2[] uv = this.myMesh.uv;
        i = 0;
        while (i < uv.Length)
        {
            Vector2 myuv = newUV + (adv * i);
            uv[i] = uv[i] + myuv;
            ++i;
        }
        this.myMesh.uv = uv;
    }

    //Stauffer - seems unused
    public virtual void ReleaseRidge()
    {
        Debug.Log("Destroying in ReleaseRidge");
        //				gameObject.Destroy(myMeshObject.gameObject);
    }

}